lib/rate-limiter.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const _get = require('lodash.get');
+
+exports.register = (defaultRate, routes) => {
+  const rateTable = {};
+
+  routes.forEach((route) => {
+    if (!(route.method === 'get' || route.method === 'post' || route.method === 'delete')) {
+      return;
+    }
+
+    if (_get(route, 'settings.plugins.rateLimit.enabled') === false) {
```

Comment:
From what I saw, there isn't necessarily a `plugin` key in settings if there's no plugin settings specified in the route declaration. We'd have to check each successive key if it's in the previous dictionary (`if plugins in settings && rateLimit in plugins && key in rateLimit`) - lodash helps shorten that.